src/routes/bounce_activities_page.jsx

```diff
@@ -0,0 +1,227 @@
+    if (activities) {
+      this.setState({
+        isFetching: false,
+        activityLog: activities.reverse(),
+      });
+    }
+  }
+
+  logout() {
+    const { history } = this.props;
+    localStorage.clear();
+    history.push("/");
+  }
+
+  handleModalClose(e) {
+    const { id } = e.currentTarget;
+    this.setState({
+      [id]: false,
+      isInvalidInput: false,
+      selectedRule: {},
+    });
+  }
+
+  updateSearchToken(e) {
+    this.setState({
+      searchToken: e.target.value.toLowerCase(),
+    });
+  }
+
+  updateSearchCategory(e) {
+    this.setState({
+      searchCategory: e.value.toLowerCase(),
+    });
+  }
+
+  filterRules(rules) {
+    const { searchToken } = this.state;
+    return rules.filter(
+      rule =>
+        rule.bounce_action.toLowerCase().includes(searchToken) ||
+        rule.description.toLowerCase().includes(searchToken)
+    );
+  }
+
+  paginate(activityLog) {
+    const { currentPageIndex, rulesToShow } = this.state;
+    const ruleStartIndex = (currentPageIndex - 1) * rulesToShow;
+    const ruleEndIndex =
+      (currentPageIndex - 1 * currentPageIndex + rulesToShow) *
+      currentPageIndex;
+    return activityLog.slice(ruleStartIndex, ruleEndIndex);
+  }
+
+  updatePageIndex(e) {
+    const newIndex = parseInt(e.currentTarget.getAttribute("value"), 10);
+    this.setState(prevState => {
+      const isPageIndexUpdated = prevState.currentPageIndex !== newIndex;
+      return {
+        currentPageIndex: isPageIndexUpdated
+          ? newIndex
+          : prevState.currentPageIndex,
+      };
+    });
+  }
+
+  handlePrevClicked() {
+    this.setState(prevState => ({
+      currentPageIndex: prevState.currentPageIndex - 1,
+    }));
+  }
+
+  handleNextClicked() {
+    this.setState(prevState => ({
+      currentPageIndex: prevState.currentPageIndex + 1,
+    }));
+  }
+
+  isDuplicate(searchCategory, searchToken) {
+    let found = false;
+    const { filterOptions } = this.state;
+    filterOptions.filter(filter => {
+      if (
+        filter.searchCategory === searchCategory &&
+        filter.searchToken === searchToken
+      ) {
+        found = true;
+      }
+      return false;
+    });
+    return found;
+  }
+
+  addFilter() {
+    const { searchCategory, searchToken } = this.state;
+    if (!searchCategory || !searchToken) {
+      this.setState({
+        isValidFilter: false,
+      });
+      return;
+    }
+    if (!this.isDuplicate(searchCategory, searchToken)) {
+      this.setState(prevState => ({
+        isValidFilter: true,
+        filterOptions: [
+          ...prevState.filterOptions,
+          { searchCategory, searchToken },
+        ],
+        searchToken: "",
+      }));
+    } else {
+      this.setState({
+        isValidFilter: false,
+      });
+    }
+  }
+
+  removeFilter(filter) {
+    const { filterOptions } = this.state;
+    const newFilter = [...filterOptions];
+    const index = newFilter.indexOf(filter);
+    newFilter.splice(index, 1);
```

Comment:
Let's not use splice and indexOf (because objects won't work exactly the same) but the .filter way I put in the other PR